### PR TITLE
Add expanding sidebar menus

### PR DESCRIPTION
**Pull request by radotreyes**
Attempting to resolve list items 3, 4, and 5 of #102. 

**EDIT**: Issue is resolved, should be ready for review and merge!

~~Feature is working as expected, but having issues rewriting tests on `Sidebar` (`npm run test` throws a warning):~~

image

~~Shallow rendering `Sidebar` seems to be still attempting to render its child components. Think it's related to the fact that I wrote a snapshot test. Still a WIP because of this.~~

### CODE REVIEW

File: src/__tests__/DrawerMenu.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import DrawerMenu from '../components/DrawerMenu';
+
+it('should render without breaking', () => {
+  const comp = shallow(<DrawerMenu id="0" to="" title="" nest={1} />);
+  expect(comp).toMatchSnapshot();
```
Comment (radotreyes) [inline src/__tests__/DrawerMenu.js]: I think the fact that I chose to write a snapshot test is the reason that this particular test throws a warning. Doing more research on the topic.

File: src/components/DrawerMenu/DrawerMenu.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import { Link } from 'react-router-dom';
+import { ListItem, DrawerMenuParent, DrawerMenuChild } from './styled';
+
+const DrawerMenu = props => {
+  const { children, id, to, title, nest } = props;
+  return (
+    <Toggle>
+      {({ show, toggle }) => (
+        <DrawerMenuParent>
+          <ListItem key={id} nest={nest} onClick={toggle}>
```
Comment (austintackaberry) [inline src/components/DrawerMenu/DrawerMenu.js]: Looks like it's just a matter of moving this `onClick={toggle}` from the `ListItem` to the `span` below

**Comment by radotreyes** [comment]
I agree with your third point, I don't really think it's necessary for us to have the X button there on desktop, and on mobile it sure does make sense to be able to just tap on the screen to close it. 

That being said, I think the first thing someone would do if we didn't have an X button would be to swipe the sidebar away. I think that because we can't support swiping/gestures (or can we?), it might actually take away from the design to not include that button, since it isn't so intuitive to tap a now smaller section of the screen to close the sidebar.

I'm about to push fixes to the two bugs you pointed out, thanks for catching them!

**Comment by austintackaberry** [comment]
Looks like the build is failing, probably just lint errors

**Comment by radotreyes** [comment]
My mistake, should be good now!

**Pull request merged**